source/code/plugins/change_tracking_runner.rb

```diff
@@ -33,67 +32,45 @@ def self.transform_and_wrap()
 			file = File.open(CHANGE_TRACKING_FILE, "rb")
 			xml_string = file.read; nil # To top the output to show up on STDOUT.
 
+			previousSnapshot = ChangeTracking.getHash(CHANGE_TRACKING_STATE_FILE)
+                        previous_inventory_checksum = {}
+                        begin
+                          if !previousSnapshot.nil?
+                            previous_inventory_checksum = JSON.parse(previousSnapshot["PREV_HASH"])
+                          end
+                        rescue 
+			@@log.warn ("Error parsing previous hash file")
+                             previousSnapshot = nil
+                        end
 			#Transform the XML to HashMap
 			transformed_hash_map = ChangeTracking.transform(xml_string, @@log)
-			output = ChangeTracking.wrap(transformed_hash_map, @hostname, time)
-			hash = Digest::SHA256.hexdigest(output.to_json)
+                        current_inventory_checksum = ChangeTracking.computechecksum(transformed_hash_map)
+                        changed_checksum = ChangeTracking.comparechecksum(previous_inventory_checksum, current_inventory_checksum)
+                        transformed_hash_map_with_changes_marked = ChangeTracking.markchangedinventory(changed_checksum, transformed_hash_map)
 
-			previousSnapshot = getHash()
+			output = ChangeTracking.wrap(transformed_hash_map_with_changes_marked, @hostname, time)
+			hash = current_inventory_checksum.to_json
 
 			# If there is a previous hash
 			if !previousSnapshot.nil?
 				# If you need to force send
 				if force_send_run_interval > 0 and 
 					Time.now.to_i - previousSnapshot[LAST_UPLOAD_TIME].to_i > force_send_run_interval
-					setHash(hash, Time.now)
-				# If the content changed.
-				elsif hash != previousSnapshot[PREV_HASH]
-					setHash(hash, Time.now)
+					ChangeTracking.setHash(hash, Time.now,CHANGE_TRACKING_STATE_FILE)
 				else
-					return {}
+					ChangeTracking.setHash(hash, Time.now, CHANGE_TRACKING_STATE_FILE)
 				end
 			else # Previous Hash did not exist. Write it
 				# and the return the output.
-				setHash(hash, Time.now)
+				ChangeTracking.setHash(hash, Time.now, CHANGE_TRACKING_STATE_FILE)
 			end
 			return output
 		else
-			@@log.info ("The change tracking inventory file does not exist and runner will be disabled.")
+			@@log.warn ("The ChangeTracking File does not exists. Make sure it is present at the correct")
```

Comment:
Looks like this hasn't been rebased to include Narine's message change: https://github.com/Microsoft/OMS-Agent-for-Linux/commit/f282c340fe8950165103f701cc24835811ea3bd7